src/pages/DelegatoryValidator/Components/TimeDurationIntervalBar.tsx

```diff
@@ -12,37 +14,66 @@ export default function TimeDurationIntervalBar({
     return null;
   }
 
+  const [remainingTime, setRemainingTime] = useState<moment.Duration>();
+  const [percentageComplete, setPercentageComplete] = useState<number>(0);
+
   // the beginning of the unlock cycle
   const startTime = parseTimestamp(timestamp.toString()).subtract(30, "days");
 
   // the end of the unlock cycle
   const unlockTime = parseTimestamp(timestamp.toString());
 
-  // the remaining time of the unlock cycle
-  const now = moment();
-  const remainingTime = moment.duration(
-    unlockTime.valueOf() - now.valueOf(),
-    "milliseconds",
-  );
+  useEffect(() => {
+    const refresh = () => {
```

Comment:
i change it back to using refresh cuz i realized with directly using the `setInterval` the first time it will not be ready but we need to wait like 1 min afterwards for it to take effect. creating a function called `refresh` and call it right after would trigger it right away. as suggested in this [post](https://stackoverflow.com/questions/6685396/execute-the-setinterval-function-without-delay-the-first-time).